tools/autograd/templates/python_torch_functions.cpp

```diff
@@ -534,4 +497,91 @@ void initTorchFunctions(PyObject* module) {
   }
 }
 
+PyObject* handle_torch_function(PythonArgs &r, PyObject* args, PyObject* kwargs, PyTypeObject &torch_api) {
+  PyObject* torch_api_function =
+    PyObject_FastGetAttrString((PyObject*)&torch_api, const_cast<char*>(r.get_func_name().data()));
+  TORCH_INTERNAL_ASSERT(torch_api_function != NULL, "torch API function must exist");
+  PyObject* ret = nullptr;
+  for (auto arg : r.overloaded_args) {
+    PyObject* torch_function = PyObject_FastGetAttrString(arg, "__torch_function__");
```

Comment:
It sure looks to me like you're leaking `torch_function`... (pssssst! Stop manually refcounting! Use the RAII structs! Your reviewers will thank you!!!)

Reply:
> (pssssst! Stop manually refcounting! Use the RAII structs! Your reviewers will thank you!!!)

Sorry, it wasn't clear to me from your comment yesterday that you really wanted me to use pybind11, I'll update this today to use that.

Reply:
(I stopped reviewing after seeing this)